core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -79,6 +79,7 @@ private[spark] abstract class Task[T](
     SparkEnv.get.blockManager.registerTask(taskAttemptId)
     context = new TaskContextImpl(
       stageId,
+      stageAttemptId, // stageAttemptId and stageAttemptNumber are semantically equal
```

Comment:
How much work we need to rename the internal `stageAttemptId` to `stageAttemptNumber`?

Reply:
The modification may not be too much (100+ occurrences in 20+ files), however it may break eventLog's JsonProtocol backward compatibility(not sure)..

@squito you may have more knowledge on this since you introduced `stageAttemptId`.

Reply:
ah, so `stageAttemptId` is already exposed in developer API, we can't change it.